### PR TITLE
Contract monitor volatility and error sign direction

### PR DESCRIPTION
**Pull request by chrismaree**
This PR adds the error directly of the price movement to the volatility monitor and synthetic peg monitor. 

close: #1509

### CODE REVIEW

File: financial-templates-lib/logger/SpyTransport.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// This transport enables unit tests to validate values passed Winston using a Sinon Spy.
+// This transport enables unit tests to validate values passed to Winston using a Sinon Spy.
```
Comment (chrismaree) [inline financial-templates-lib/logger/SpyTransport.js:2]: cleaned up this comment. unrelated to PR

File: monitors/SyntheticPegMonitor.js [hunk 2]
```diff
@@ -218,45 +218,55 @@ class SyntheticPegMonitor {
...
 
-  // Takes in two big numbers and returns the error between them.
-  // calculated using: δ = | (observed - expected) / expected |
-  // For example an observed price of 1.2 with an expected price of 1.0 will return | (1.2 - 1.0) / 1.0 | = 0.20
-  // This is equivalent of a 20 percent absolute deviation between the numbers.
+  // Takes in two big numbers and returns the error between them. using: δ = (observed - expected) / expected
+  // For example an observed price of 1.2 with an expected price of 1.0 will return (1.2 - 1.0) / 1.0 = 0.20
+  // This is equivalent of a 20 percent deviation between the numbers.
+  // Note that this logger can return negative error if the deviation is in a negative direction.
   _calculateDeviationError(observedValue, expectedValue) {
```
Comment (chrismaree) [inline monitors/SyntheticPegMonitor.js:225]: this function no longer returns an absolute value but rather a directional percentage error.

File: monitors/SyntheticPegMonitor.js [hunk 3]
```diff
@@ -218,45 +218,55 @@ class SyntheticPegMonitor {
...
+  // until `mostRecentTime`. Returns volatility as (max - min)/min %. Also Identifies the direction volatility movement.
   _calculateHistoricalVolatility(pricefeed, mostRecentTime, lookback) {
     // Set max and min to latest price to start.
     let min = pricefeed.getHistoricalPrice(mostRecentTime);
     let max = min;
     if (!min || !max) return null;
 
+    // Store the timestamp of the max and min value to infer the direction of the movement over the interval.
+    let maxTimestamp = 0,
+      minTimestamp = 0;
```
Comment (nicholaspai) [inline monitors/SyntheticPegMonitor.js:242]: Why not use a `let` for `minTimestamp`?

File: monitors/SyntheticPegMonitor.js [hunk 4]
```diff
@@ -218,45 +218,55 @@ class SyntheticPegMonitor {
...
-  // until `mostRecentTime`. Returns volatility as (max - min)/min %.
+  // until `mostRecentTime`. Returns volatility as (max - min)/min %. Also Identifies the direction volatility movement.
   _calculateHistoricalVolatility(pricefeed, mostRecentTime, lookback) {
     // Set max and min to latest price to start.
     let min = pricefeed.getHistoricalPrice(mostRecentTime);
     let max = min;
     if (!min || !max) return null;
 
+    // Store the timestamp of the max and min value to infer the direction of the movement over the interval.
+    let maxTimestamp = 0,
```
Comment (nicholaspai) [inline monitors/SyntheticPegMonitor.js:241]: +1 Tracking the `max` and `min` timestamps would be useful outside of this PR

File: monitors/SyntheticPegMonitor.js (hunk 3 above)
Comment (chrismaree) [reply monitors/SyntheticPegMonitor.js:242]: there is a `,` so both are `let`

**Pull request merged**